File: s11_BroadcastReciver/BroadcastReciverPermission2/app/src/main/java/com/omeryildizce/broadcastreciverpermission2/MainActivity.java
package com.omeryildizce.broadcastreciverpermission2;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void signal(View view) {
        Intent intent = new Intent("my.signal");
        sendBroadcast(intent,"my.permission");
    }
}

File: s03_DaggerHilt/HiltKotlin/app/src/main/java/com/omeryildizce/hiltkotlin/MainActivity.kt

package com.omeryildizce.hiltkotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.fragment.app.Fragment
import dagger.hilt.android.AndroidEntryPoint
import javax.inject.Inject

@AndroidEntryPoint
class MainActivity : AppCompatActivity() {
    // field injection
    @Inject
    lateinit var lars: Musicians

    @Inject
    lateinit var myClass : ClassExample
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        lars.sing()
        println( myClass.myFunction())
        println(myClass.seconfFunction())
    }
}

